pyresample/bilinear/_base.py

```diff
@@ -599,14 +599,14 @@ def get_valid_indices_from_lonlat_boundaries(
         target_geo_def, source_lons, source_lats, radius_of_influence):
     """Get valid indices from lonlat boundaries."""
     # Resampling from swath to grid or from grid to grid
-    lonlat_boundary = target_geo_def.get_boundary_lonlats()
-
-    # Combine reduced and legal values
-    return data_reduce.get_valid_index_from_lonlat_boundaries(
-        lonlat_boundary[0],
-        lonlat_boundary[1],
-        source_lons, source_lats,
-        radius_of_influence)
+    try:
+        sides_lons, sides_lats = target_geo_def.boundary().sides
+        valid_indices = data_reduce.get_valid_index_from_lonlat_boundaries(sides_lons, sides_lats,
+                                                                           source_lons, source_lats,
+                                                                           radius_of_influence)
+    except Exception:
+        valid_indices = np.ones(source_lons.size, dtype=bool)
```

Comment:
No @djhoese. This try-except is used to deal with out of Earth disk projections !!!
Before this PR, the boundary was created using `get_boundary_lonlats` (see https://github.com/pytroll/pyresample/blob/294e6ea797a9e2b0aa61c005df5ed7e4bbf359e3/pyresample/geometry.py#L276). **The method was just naively taking the image border lon/latcoordinates.**
So for the geostationary full disk and other projections with out-of-earth disk at the border, the boundary sides were all `Inf`. And inside the .`get_valid_index_from_lonlat_boundaries` if there was any invalid side coordinate, it was returning an array of ones (no data reduction done !). **This means that for full disc GEO projections we were not doing data reduction** !!!

In this PR, I replaced `get_boundary_lonlats` with `target_geo_def.boundary().sides`. So:

- if the area is geostationary, now we finally reduce data (and this should speed up stuffs !!!)
- if the area boundary has out-of-Earth locations and the flag `force_boundary_computations` is `True`, the boundary method will retrieve the actual boundary coordinate inside the area, and we will reduce data ...
- if the area boundary has out-of-Earth locations and the flag `force_boundary_computations` is `False` (to keep the old behaviour), the boundary method now fails because if all sides are `Inf` it will raise a `ValueError` within `_filter_bbox_nans`. If this happen, we return what we returned before: the `np.ones` array